pandas/tests/indexes/test_interval.py

```diff
@@ -458,6 +592,30 @@ def test_get_indexer_subintervals(self):
         expected = np.array([0, 0, 0], dtype='intp')
         tm.assert_numpy_array_equal(actual, expected)
 
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_get_indexer_subintervals_updated_behavior(self):
+
+        # a good chance this whole method needs reworking.
+
+        # target = IntervalIndex.from_breaks(np.linspace(0, 2, 5))
```

Comment:
@jorisvandenbossche the way I made this PR is I copied basically each method and wrote how I thought it should behave, but for methods like this one, I either didn't understand the original functionality or I wasn't sure how it should be updated. I think it's likely we'll want to change how this method works to be logically consistent with the updated new methods, but I'm not sure how to do that, mostly because, in this case, I'm not perfectly lucid on how get_indexer is supposed to behave.